lib/v2.recommended.yaml

```diff
@@ -0,0 +1,231 @@
+    use_function_type_syntax_for_parameters: warning
+    use_rethrow_when_possible: warning
+    use_setters_to_change_properties: warning
+    use_string_buffers: warning
+    use_to_and_as_if_applicable: warning
+
+linter:
+  rules:
+    - always_declare_return_types
+    - always_put_required_named_parameters_first
+    - always_require_non_null_named_parameters
+    - avoid_bool_literals_in_conditional_expressions
+    - avoid_field_initializers_in_const_classes
+    - avoid_function_literals_in_foreach_calls
+    - avoid_implementing_value_types
+    - avoid_js_rounded_ints
+    - avoid_null_checks_in_equality_operators
+    - avoid_positional_boolean_parameters
+    - avoid_relative_lib_imports
+    - avoid_renaming_method_parameters
+    - avoid_returning_this
+    - avoid_returning_null
+    - avoid_returning_null_for_future
+    - avoid_returning_null_for_void
+    - avoid_setters_without_getters
+    - avoid_slow_async_io
+    - avoid_types_on_closure_parameters
+    - avoid_unused_constructor_parameters
+    - cascade_invocations
+    - comment_references
+    - control_flow_in_finally
+    - curly_braces_in_flow_control_structures
+    - directives_ordering
+    - duplicate_ignore
+    - invalid_required_named_param
+    - invalid_required_positional_param
+    - invariant_booleans
+    - join_return_with_assignment
+    - literal_only_boolean_expressions
+    - omit_local_variable_types
+    - overridden_fields
+    - package_api_docs
+    - package_prefixed_library_names
+    - package_names
+    - prefer_adjacent_string_concatenation
+    - prefer_asserts_in_initializer_lists
+    - prefer_collection_literals
+    - prefer_conditional_assignment
+    - prefer_const_constructors_in_immutables
+    - prefer_const_declarations
+    - prefer_const_literals_to_create_immutables
+    - prefer_constructors_over_static_methods
+    - prefer_typing_uninitialized_variables
+    - prefer_final_fields
+    - prefer_final_in_for_each
+    - prefer_final_locals
+    - prefer_function_declarations_over_variables
+    - prefer_if_elements_to_conditional_expressions
+    - prefer_initializing_formals
+    - prefer_inlined_adds
+    - prefer_int_literals
+    - prefer_interpolation_to_compose_strings
+    - prefer_iterable_whereType
+    - prefer_mixin
+    - prefer_null_aware_operators
+    - prefer_single_quotes
+    - prefer_spread_collections
+    - prefer_void_to_null
+    - provide_deprecation_message
+    - sort_pub_dependencies
+    - sort_unnamed_constructors_first
+    - test_types_in_equals
+    - throw_in_finally
+    - type_annotate_public_apis
+    - unawaited_futures
+    - unnecessary_await_in_return
+    - unnecessary_brace_in_string_interps
+    - unnecessary_cast
+    - unnecessary_getters_setters
+    - unnecessary_lambdas
+    - unnecessary_parenthesis
+    - unnecessary_overrides
+    - unnecessary_this
+    - use_function_type_syntax_for_parameters
+    - use_rethrow_when_possible
+    - use_setters_to_change_properties
```

Comment:
Previously we discovered that this lint would introduce a lot of infractions in UI code: https://github.com/Workiva/workiva_analysis_options/issues/146#issuecomment-555679874

Maybe that's no longer the case. I take it RP hasn't had any issue with this one?

Reply:
We have this disabled in many subpackages. I'm going to turn that on everywhere and see what is going on there.

Reply:
Ok, yeah, I wouldn't say this is about UI code. 

I believe that this conflicts with avoid_setters_without_getters. We have not consistently enabled this in DPC and I could 100% be on board with not enforcing it.